### PR TITLE
[ refactor ] unify Prop/Type/Set as Univ

### LINKED ISSUE #6621: Enable K also for SSetω (like for SSet)
@jespercockx : Should this check for `SSet` be extended to include `SSetOmega`?
https://github.com/agda/agda/blob/1aaeb528e7880ea31f5d87ab9bd1918736e36a3e/src/full/Agda/TypeChecking/Rules/LHS.hs#L1362-L1364
https://github.com/agda/agda/blob/1aaeb528e7880ea31f5d87ab9bd1918736e36a3e/src/full/Agda/TypeChecking/Coverage.hs#L986-L988
Test:
```agda
{-# OPTIONS --two-level --without-K #-}

module _ where

open import Agda.Primitive

module Zero where

  data Id (A : SSet) (a : A) : A → SSet where
    refl : Id A a a

  K : {A : SSet} {a : A} (p q : Id A a a) → Id (Id A a a) p q
  K refl refl = refl

  -- passes

module Omega where

  data Id (A : SSetω) (a : A) : A → SSetω where
    refl : Id A a a

  K : {A : SSetω} {a : A} (p q : Id A a a) → Id (Id A a a) p q
  K refl refl = refl

  -- I'm not sure if there should be a case for the constructor refl,
  -- because I get stuck when trying to solve the following unification
  -- problems (inferred index ≟ expected index):
  --   a ≟ a
  -- Possible reason why unification failed:
  --   Cannot eliminate reflexive equation a = a of type A because K has
  --   been disabled.
  -- when checking that the pattern refl has type Id A a a

```

### LINKED ISSUE #6624: Suffix not working for SSet
```agda
{-# OPTIONS --two-level #-}

open import Agda.Primitive

S : SSet₁
S = SSet
```
Not in scope: `SSet₁`

### PR DESCRIPTION
**Pull request by andreasabel**
Internal refactoring of the sort representation.

Also: fix #6621, fix #6624.
- #6621
- #6624

### CODE REVIEW

**Review by jespercockx — APPROVED** [review]

File: Agda.cabal [hunk 1]
```diff
@@ -247,6 +246,9 @@ common language
 
   if impl(ghc >= 8.10)
     ghc-options:
+      -Werror
+         -- #6623: Cannot have -Werror on GHC < 8.10 because of (nameless!?) warning.
+         -- "Pattern match checker exceeded (2000000) iterations in a case alternative."
```
Comment (nad) [inline Agda.cabal:251]: Did you try to turn off the warning [`incomplete-patterns`](https://downloads.haskell.org/ghc/8.8.4/docs/html/users_guide/using-warnings.html#ghc-flag--Wincomplete-patterns) (and perhaps also [`incomplete-uni-patterns`](https://downloads.haskell.org/ghc/8.8.4/docs/html/users_guide/using-warnings.html#ghc-flag--Wincomplete-uni-patterns)) for old versions of GHC? I think that is better than shipping code that generates a warning when compiled.

Comment (nad) [reply Agda.cabal:251]: Now I also noticed that the warning appears on the "Files changed" tab of pull requests, under the heading "Unchanged files with check annotations".

Comment (andreasabel) [reply Agda.cabal:251]: I tried, adding `-Wno-warn-incomplete-patterns`, but it didn't have an effect on this warning.  I also noticed that this warning does not come with a name (unlike all other warnings), so maybe it cannot be turned off?

Comment (nad) [reply Agda.cabal:251]: Someone else [had](https://gitlab.haskell.org/ghc/ghc/-/issues/13464) a similar problem, and the solution was to disable `overlapping-patterns` as well.

Comment (andreasabel) [reply Agda.cabal:251]: Thanks for the pointer.  Fix is coming.

Comment (andreasabel) [reply Agda.cabal:251]: 0dd48ba0a93f5fd06195534cc4e2aa6cd4523ab1

**Pull request merged**